src/spikeinterface/core/base.py

```diff
@@ -329,6 +330,8 @@ def to_dict(
             Folder with numpy `npy` files containing additional information (e.g. probe in BaseRecording) and properties.
         recursive: bool
             If True, all dicitionaries in the kwargs are expanded with `to_dict` as well, by default False.
+        skip_recursive_path_modifier_warning: bool
+            If True, skip the warning that is raised when `recursive=True` and `relative_to` is not None.
```

Comment:
In which situation we would have `recursive=True` and `relative_to` not None ?
I think this option is too much detail and to_dict() is more or less internal.
So I think that either we are strict and we raise an error either we do not care but adding this flag is quite a bit heavy for the API no ?

Reply:
as you said, `to_dict` is internal so we can keep it in my opinion